### PR TITLE
Feature/main feature

### PR DESCRIPTION
**Pull request by pacyL2K19**
# The aim of this PR 
- It contains the dom-manipulation for this project 
- Implementation of objects for the main parts of the project 
- Setup of linters 
- Style sheet of the project

### CODE REVIEW

File: src/DomFunctionality.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+const newProjectDom = (() => {
+  const addProjectBtn = document.querySelector('#add-project');
+  const submitProjectBtn = document.getElementById('submit-project');
+  const cancelSubmitProjectBtn = document.getElementById('cancel-submit-project');
+  const submitProjectWindow = document.querySelector('.submit-project-window');
+  const _nameField = document.getElementById('project-name');
+  const _descriptionField = document.getElementById('project-description');
+  const alertSubmitProject = document.getElementById('alert-submit-project');
+
+  function clear() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    _nameField.value = '';
+    _descriptionField.value = '';
+  }
+
+  function hide() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    clear();
+    submitProjectWindow.style.display = 'none';
+  }
+
+  function show() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    editTaskDom.hide();
+    newTaskDom.hide();
+    submitProjectWindow.style.display = 'block';
+  }
+
+  function submitProject() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    show,
+    hide,
+    submitProject,
+  };
+})();
+
+const displayHeader = (() => {
+  const headerContainer = document.querySelector('.active-project-container');
+
+  function show() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    headerContainer.style.display = 'flex';
+  }
+
+  function hide() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+  const addTaskBtn = document.querySelector('.add-task');
+  const addTaskWindow = document.getElementById('submit-task-window');
+  const submitTaskBtn = document.getElementById('submit-task');
+  const cancelTaskBtn = document.getElementById('cancel-submit-task');
+  const _nameField = document.getElementById('task-name');
+  const _dueDateField = document.getElementById('task-due-date');
+  const _priorityField = document.getElementById('priority');
+  const alertSubmitTask = document.getElementById('alert-submit-task');
+
+  function show() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    editTaskDom.hide();
+    newProjectDom.hide();
+    addTaskWindow.style.display = 'block';
+  }
+
+  function hide() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+    clear();
+    addTaskWindow.style.display = 'none';
+  }
+
+  function clear() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+    _nameField.value = '';
+    _dueDateField.value = '';
+    _priorityField.value = '';
+  }
+
+  function submitTask() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+  const editTaskWindow = document.getElementById('edit-task-window');
+  const updateTaskBtn = document.getElementById('update-task');
+  const cancelUpdateBtn = document.getElementById('cancel-update-task');
+  const _nameField = document.getElementById('edit-task-name');
+  const _dueDateField = document.getElementById('edit-task-due-date');
+  const _priorityField = document.getElementById('edit-priority');
+  const alertEditTask = document.getElementById('alert-edit-task');
+  let selectedTask;
+
+  function show(index) {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+    newProjectDom.hide();
+    newTaskDom.hide();
+    editTaskWindow.style.display = 'block';
+    selectedTask = index;
+  }
+
+  function hide() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+    clear();
+    editTaskWindow.style.display = 'none';
+  }
+
+  function clear() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+    _nameField.value = '';
+    _dueDateField.value = '';
+    _priorityField.value = '';
+  }
+
+  function updateTask() {
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Use arrow functions instead of the function keyword

File: src/DynamicListener.js [hunk 15]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable func-names */
+/* eslint-disable import/no-cycle */
+import {
+  deleteProject, setActiveProject, deleteTask, updateTaskStatus,
+} from './Logic';
+import { editTaskDom } from './DomFunctionality';
+
+function addProjectListeners() {
```
Comment (rindrajosia) [inline src/DynamicListener.js:8]: Use arrow functions instead of the function keyword

File: src/DynamicListener.js [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+  deleteProjectBtn.forEach((project) => {
+    project.addEventListener('click', function () {
+      const index = this.parentNode.getAttribute('data-index');
+      deleteProject(index);
+    });
+  });
+}
+
+function addTasksListeners() {
```
Comment (rindrajosia) [inline src/DynamicListener.js:27]: Use arrow functions instead of the function keyword

File: src/Render.js [hunk 17]
```diff
@@ -0,0 +1,105 @@
+/* eslint-disable import/no-cycle */
+import { allProjects } from '.';
+import { addProjectListeners, addTasksListeners } from './DynamicListener';
+import { activeProject } from './Logic';
+
+function renderProjects() {
```
Comment (rindrajosia) [inline src/Render.js]: Use arrow functions instead of the function keyword

File: src/Render.js [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+    singleProjectContainer.appendChild(projectTitle);
+    singleProjectContainer.appendChild(deleteProjectBtn);
+    projectList.appendChild(singleProjectContainer);
+  });
+
+  addProjectListeners();
+}
+
+function renderHeader() {
```
Comment (rindrajosia) [inline src/Render.js]: Use arrow functions instead of the function keyword

File: src/Render.js [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+
+    const activeDescription = document.createElement('h5');
+    activeDescription.innerText = activeProject().description;
+
+    activeProjectDetails.appendChild(activeTitle);
+    activeProjectDetails.appendChild(activeDescription);
+  }
+}
+
+function renderTasks() {
```
Comment (rindrajosia) [inline src/Render.js]: Use arrow functions instead of the function keyword

File: src/DomFunctionality.js [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+  const alertSubmitProject = document.getElementById('alert-submit-project');
+
+  function clear() {
+    _nameField.value = '';
+    _descriptionField.value = '';
+  }
+
+  function hide() {
+    clear();
+    submitProjectWindow.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  function show() {
+    editTaskDom.hide();
+    newTaskDom.hide();
+    submitProjectWindow.style.display = 'block';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+  function submitProject() {
+    if (_nameField.value !== '' && _descriptionField.value !== '') {
+      const name = _nameField.value;
+      const description = _descriptionField.value;
+
+      createNewProject(name, description);
+      clear();
+      hide();
+    } else {
+      alertSubmitProject.style.display = 'block';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+      setTimeout(() => {
+        alertSubmitProject.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+    hide,
+    submitProject,
+  };
+})();
+
+const displayHeader = (() => {
+  const headerContainer = document.querySelector('.active-project-container');
+
+  function show() {
+    headerContainer.style.display = 'flex';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  function hide() {
+    headerContainer.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 26]
```diff
@@ -0,0 +1,184 @@
...
+  const cancelTaskBtn = document.getElementById('cancel-submit-task');
+  const _nameField = document.getElementById('task-name');
+  const _dueDateField = document.getElementById('task-due-date');
+  const _priorityField = document.getElementById('priority');
+  const alertSubmitTask = document.getElementById('alert-submit-task');
+
+  function show() {
+    editTaskDom.hide();
+    newProjectDom.hide();
+    addTaskWindow.style.display = 'block';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 27]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  function hide() {
+    clear();
+    addTaskWindow.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 28]
```diff
@@ -0,0 +1,184 @@
...
+  function submitTask() {
+    if (_nameField.value !== '' && _dueDateField.value !== '' && _priorityField.value !== '') {
+      const name = _nameField.value;
+      const dueDate = _dueDateField.value;
+      const priority = _priorityField.value;
+
+      createNewTask(name, dueDate, priority);
+      hide();
+    } else {
+      alertSubmitTask.style.display = 'block';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 29]
```diff
@@ -0,0 +1,184 @@
...
+      setTimeout(() => {
+        alertSubmitTask.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 30]
```diff
@@ -0,0 +1,184 @@
...
+  const _nameField = document.getElementById('edit-task-name');
+  const _dueDateField = document.getElementById('edit-task-due-date');
+  const _priorityField = document.getElementById('edit-priority');
+  const alertEditTask = document.getElementById('alert-edit-task');
+  let selectedTask;
+
+  function show(index) {
+    newProjectDom.hide();
+    newTaskDom.hide();
+    editTaskWindow.style.display = 'block';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 31]
```diff
@@ -0,0 +1,184 @@
...
+    selectedTask = index;
+  }
+
+  function hide() {
+    clear();
+    editTaskWindow.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 32]
```diff
@@ -0,0 +1,184 @@
...
+  function updateTask() {
+    if (_nameField.value !== '' && _dueDateField.value !== '' && _priorityField.value !== '') {
+      const name = _nameField.value;
+      const dueDate = _dueDateField.value;
+      const priority = _priorityField.value;
+
+      editTask(selectedTask, name, dueDate, priority);
+      hide();
+    } else {
+      alertEditTask.style.display = 'block';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

File: src/DomFunctionality.js [hunk 33]
```diff
@@ -0,0 +1,184 @@
...
+      setTimeout(() => {
+        alertEditTask.style.display = 'none';
```
Comment (rindrajosia) [inline src/DomFunctionality.js]: Keep all the styling logic in CSS files

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Good design :+1: 

To improve :factory_worker: [REQUIRED]:

**- Webpack :**

- [x] I got an error (Error: Cannot find module 'webpack-cli/bin/config-yargs') when I use `npm run start` command. Fortunately `npm run build` works. I suggest that you fix it.


**- Application :**
 - [x] I can see that you are trying to use localStorage however, data are not saved in the sessions. Please fix it.
 - [x] I got an error (Uncaught TypeError: f is not a function) when I create, delete and edit projects and tasks. It will be perfect if you work on it.


**- Task :** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/todo_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/todo-list))
     
- [x]  It has a description.

**- Additional Microverse Requirements :**

- [x] Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store, .vscode ) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)

**- ES6 syntax :** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/))

- [x]  Use `arrow functions` instead of `the function keyword`

Nice to have:

- [ ] Please use select option for priority (task)

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Comment by pacyL2K19** [comment]
## Local Storage
It works on my side, and didn't find any reason to change, kindly tell me if it stills doesn't work for you

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this project. 
### Changes made 🥳 🥳 🥳 
- Fixed error with running `npm start`
- Use of arrow syntax
- Unnecessary files/folders added to `.gitignore`
### Changes to make 🙅🏿 🙅🏿 🙅🏿 
- Todos are not persisted to local storage. On reload all tasks disappear and I get an error: `Uncaught TypeError: ___WEBPACK_IMPORTED_MODULE_0__.setLocalMemory is not a function`


### Martin Afani
#### Slack: whiz25

**Comment by pacyL2K19** [comment]
Removed warnings in the console

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

Let's address the comment about using [LocalStorage](https://www.w3schools.com/jsref/prop_win_localstorage.asp) mentioned from the previous reviewer;
Todos are not persisted to local storage.

Screenshot from Odin project specification page;
image


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi @pacyL2K19, congratulations on this project :1st_place_medal: The localStorage functionality is now implemented :fireworks: We can refresh the browser and we still have our data in browser :trophy: Awesome

I hereby approve the project, do not forget to submit the project completion form :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**